pig/src/main/java/com/netflix/iceberg/pig/SchemaUtil.java

```diff
@@ -64,9 +60,25 @@ private static ResourceFieldSchema convert(Type type) throws IOException {
     ResourceFieldSchema result = new ResourceFieldSchema();
     result.setType(convertType(type));
 
+    if(!type.isPrimitiveType()) {
+      ResourceSchema nestedSchema = new ResourceSchema();
+      nestedSchema.setFields(convertFields(type.asNestedType().fields()));
```

Comment:
If the input type is a list[long] say, then we will create a bag and add a element [of type long] to it without first wrapping it inside a tuple.

I confirmed this behavior using this test case
` @Test
  public void testLongInBag() throws IOException {
    Schema icebergSchema = new Schema(
        optional(
            1, "nested_list",
            MapType.ofOptional(
                2, 3,
                StringType.get(),
                ListType.ofRequired(5, LongType.get()))));
    SchemaUtil.convert(icebergSchema);
  }`

Also, it seems that this functionality of converting complex iceberg schema to Pig schema  is being done in 2 places. Once in `convert(com.netflix.iceberg.types.Type)` and the other in `convertComplex(Types.NestedField field)` .  Is there scope in simplifying this?